Dockerfile

```diff
@@ -1,9 +1,5 @@
 # Simple Dockerfile to run all the scripts on a base Ubuntu image
-FROM ubuntu:groovy-20200921
-
-# ENV
-ARG PAT_ARG
-ENV PAT=$PAT_ARG
+FROM ubuntu:22.04
```

Comment:
The main thing we will need to check before merging this if the change in compiler version affects the bootloader.
Not so worry about it changing the "behaviour" and it is unlikely it will introduced bugs (unless maybe there is undefined behaviour that we don't know about), but it's mostly about the binary size as we have limited allocated flash space.

- Ubuntu 20.10 Groovy had gcc-arm-none-eabi `15:9-2019-q4-0ubuntu1`
    - https://packages.ubuntu.com/groovy/allpackages
- Ubuntu 21.10 Impish has gcc-arm-none-eabi `15:9-2019-q4-0ubuntu2`
    - https://packages.ubuntu.com/impish/gcc-arm-none-eabi
- Ubuntu 22.04 Jammy has gcc-arm-none-eabi `15:10.3-2021.07-4`
    - https://packages.ubuntu.com/jammy/gcc-arm-none-eabi

So the safest upgrade would be Ubuntu 21.10, but that only has a few months of support left.

I'm not even too sure what happened with the SDK 17 PR and draft releases, so we would need to clarify that as well to make sure we know what ends up in the CODAL repo.

I'd be happy to leave this open and do the checks on the next bootloader update. If you prefer the PR to be merge sooner than that, then I'd have to reserve a bit of time to look into this.